lib/flux/app-state.js

```diff
@@ -39,6 +39,7 @@ export const actionMap = new ActionMap({
     revision: null,
     showTrash: false,
     listTitle: 'All Notes',
+    notesLoaded: false,
```

Comment:
Did we consider starting `notes: null`?

There's a chance to indicate whether or not we've loaded using this _tri-state_ value and that could preclude the need for creating a second value intrinsically bound to the same data.

```js
const noteListStatus = notes =>
  null === notes ? 'Loading…' :
  0 === notes.length ? 'No Notes' :
  'Has notes'
```

Reply:
Just an explanation: I usually try to find ways of gleaning more information from a value without adding additional flags or meta as they tend to get out of sync and because people tend to zoom-in on the value they think they need as soon as they find a value that seems to work, so for dev communication purposes I have found the tai-state value quite useful when there is a separate unknown or pending state from the empty state.

Reply:
Making the default value for notes null causes some plumbing issues. Setting the default to null doesn't do much. It immediately gets overwritten to an array. Changing the way this works I think is outside the scope of this PR.

Reply:
A month or so ago I spent several hours trying to get app state to function as you describe and I created more bugs than I was solving.

Reply:
ah interesting. I wondered what would break if we made the change.

where does it get immediately overwritten? on `authChanged` I can see we'd have to set it `null` there too.

I wouldn't question this further other than I recognize that we're adding new stuff into the app and doing so with two variables that are storing some of the same data, hypothetically.

I'll do some quick checks in a running copy locally and then report back.

Nice work, and I did at one point see your old state PR. state here in general is in a place it shouldn't be. long-term we can trim it way down